src/dune_engine/build_system.ml

```diff
@@ -6,13 +6,23 @@ module Progress = struct
   type t =
     { number_of_rules_discovered : int
     ; number_of_rules_executed : int
+    ; number_of_rules_failed : int
     }
 
-  let equal { number_of_rules_discovered; number_of_rules_executed } t =
+  let equal
+      { number_of_rules_discovered
+      ; number_of_rules_executed
+      ; number_of_rules_failed
+      } t =
     Int.equal number_of_rules_discovered t.number_of_rules_discovered
     && Int.equal number_of_rules_executed t.number_of_rules_executed
+    && Int.equal number_of_rules_failed t.number_of_rules_failed
```

Comment:
One annoying side effect of this change is that every failure will send an unnecessary update message to RPC. Not a big deal, but something to keep in mind.

Reply:
Isn't `number_of_rules_failed` counter updated in sync with `number_of_rules_executed`?

Reply:
I suppose so. Then this wouldn't be a problem.